### PR TITLE
RTL Conversion - NxStatefulAlert - RSC-1343

### PR DESCRIPTION
**Pull request by avnvbhatta**
JIRA: https://issues.sonatype.org/browse/RSC-1343

### CODE REVIEW

**Comment by rpokorny** [comment]
This seems like it duplicates the `NxAlert` tests which is necessary as we discussed, but I don't see anything testing the stateful behavior or the action of the close button.

**Comment by avnvbhatta** [comment]
> This seems like it duplicates the `NxAlert` tests which is necessary as we discussed, but I don't see anything testing the stateful behavior or the action of the close button.

I assumed that by modifying [this part of the test](https://github.com/sonatype/sonatype-react-shared-components/pull/906/files#diff-7c044be1260d9d2488ddb8dcba0ed2ab360589f20c9203b4cc1a131c9dd06d07R58-R67) so that the alerts render the close button without specifying the `onClose` prop and clicking on the button would remove it from the DOM would suffice. 

What else do you think needs to be added?

**Comment by rpokorny** [comment]
> I assumed that by modifying [this part of the test](https://github.com/sonatype/sonatype-react-shared-components/pull/906/files#diff-7c044be1260d9d2488ddb8dcba0ed2ab360589f20c9203b4cc1a131c9dd06d07R58-R67) so that the alerts render the close button without specifying the onClose prop and clicking on the button would remove it from the DOM would suffice.

What else do you think needs to be added?

Oh OK, that's closer than I had noticed.  I think that just needs to be adjusted so that it asserts the disappearance of the whole modal, not just the close button, and the name of the test could be improved.

**Comment by avnvbhatta** [comment]
> > I assumed that by modifying [this part of the test](https://github.com/sonatype/sonatype-react-shared-components/pull/906/files#diff-7c044be1260d9d2488ddb8dcba0ed2ab360589f20c9203b4cc1a131c9dd06d07R58-R67) so that the alerts render the close button without specifying the onClose prop and clicking on the button would remove it from the DOM would suffice.
> > What else do you think needs to be added?
> 
> Oh OK, that's closer than I had noticed. I think that just needs to be adjusted so that it asserts the disappearance of the whole alert, not just the close button, and the name of the test could be improved.

Updated here: dcb51074b5c1d406a71033ab33b4c5fbb429cc2c

File: lib/src/components/NxAlert/stateful/__tests__/NxStatefulAlert.test.tsx [hunk 1]
```diff
@@ -4,56 +4,158 @@
...
+    it('renders a Close button even though the onClose prop is not specified', async function() {
+      const user = userEvent.setup(),
+          view = quickRender(),
+          closeBtn = view.getByRole('button', { name: 'Close' });
+
+      expect(view.container.querySelector('.nx-alert')).toBeInTheDocument();
+      expect(closeBtn).toBeInTheDocument();
+
+      await user.click(closeBtn);
+      expect(view.container.querySelector('.nx-alert')).not.toBeInTheDocument();
```
Comment (rpokorny) [inline lib/src/components/NxAlert/stateful/__tests__/NxStatefulAlert.test.tsx:67]: The `nx-alert` class is an implementation detail. IMO a better approach would be to check that the container element is empty.

File: lib/src/components/NxAlert/stateful/__tests__/NxStatefulAlert.test.tsx [hunk 2]
```diff
@@ -4,56 +4,158 @@
...
+      expect(renderEl()).toHaveAttribute('aria-atomic', 'true');
+    });
+
+    it('passes any other props to the top-level element', function() {
+      const el = renderEl({ id: 'foo', title: 'bar' });
+      expect(el).toHaveAttribute('id', 'foo');
+      expect(el).toHaveAttribute('title', 'bar');
+    });
+
+    it('renders a Close button even though the onClose prop is not specified', async function() {
```
Comment (rpokorny) [inline lib/src/components/NxAlert/stateful/__tests__/NxStatefulAlert.test.tsx:58]:
```suggestion
    it('renders a Close button that removes the alert from the DOM when clicked', async function() {
```

File: lib/src/components/NxAlert/stateful/__tests__/NxStatefulAlert.test.tsx (hunk 1 above)
Comment (avnvbhatta) [reply lib/src/components/NxAlert/stateful/__tests__/NxStatefulAlert.test.tsx]: > The `nx-alert` class is an implementation detail. IMO a better approach would be to check that the container element is empty.

Changed here: c8e92ba8a07092dab3b8bb323be2bb8adb5b656b

**Review by rpokorny — APPROVED** [review]

**Pull request merged**